develop/tutorials/articles/02-from-liferay-6-to-liferay-7/07-upgrading-plugins-to-liferay-7/07-upgrading-application-upgrade-and-verifier-processes.markdown

```diff
@@ -0,0 +1,426 @@
+    instances, which in this example contains two steps:
+
+        new UpgradeAdminPortlets()
+        new UpgradePortletSettings(_settingsFactory)
+
+    You can use this same process to upgrade your 6.2 upgrade processes to use
+    the new framework.
+
+5.  Remove the logger code. It should look similar to the following pattern:
+
+    private static final Log _log = logFactoryUtil.getLog(
+        UpgradeProcess_1_0_0.class);
+
+    Log code is inside of the framework, so upgrade steps do not need to log
+    their operations.
+
+6.  Finally, use the `@Reference` annotation to reference the services that you
+    need for the upgrade. For example, here is a reference to the
+    `companyLocalService` for the [PrivateMessagingServiceUpgrade.java](https://github.com/liferay/liferay-portal/blob/master/modules/apps/social-private-messaging/social-private-messaging-service/src/main/java/com/liferay/social/privatemessaging/internal/upgrade/PrivateMessagingServiceUpgrade.java)
+    class:
+
+	@Reference(unbind = "-")
+	protected void setCompanyLocalService(
+		CompanyLocalService companyLocalService) {
+
+		_companyLocalService = companyLocalService;
+	}
+
+With that, your application upgrade processes are upgraded!
+
+For your convienience, a summary of the steps for upgrading your application
+upgrade process are outlined in the table below for reference:
+
+<style>
+.lego-image {
+	max-height: 100%;
+	max-width: 100%;
+}
+.max-960 {
+	margin: 0 auto;
+	max-width: 960px;
+}
+.no-max
+.max-960 {
+	max-width: none;
+}
+.metadata-guidelines-table td {
+	border-bottom: 1px solid;
+	border-top: 1px solid;
+	padding: 10px;
+}
+.table-header {
+	font-weight: bold;
+}
+.table-header.second {
+	width: 70%;
+}
+.left-header {
+	border-right: 1px solid;
+}
+</style>
+<div class="lego-article metadata-guidelines-table" id="article-33460946">
+<div class="lego-article-content max-960">
+<div class="aui-helper-clearfix lego-section section-1" >
+<div class="aui-w100 block-1 content-column lego-block" >
+<div class="content-column-content">
+<table>
+	<thead>
+		<td class="table-header left-header">
+			File
+		</td>
+		<td class="table-header second">
+			Required Changes
+		</td>
+	</thead>
+	<tbody>
+		<tr>
+			<td class="table-header left-header">
+				build.gradle File
+			</td>
+			<td class="">
+				<ul>
+					<li>
+						Add the `portal-upgrade` dependency.
+					</li>
+				</ul>
+			</td>
+		</tr>
+		<tr>
+			<td class="table-header left-header">
+			bnd.bnd file
+			</td>
+			<td class="">
+				<ul>
+					<li>
+						Add Liferay-Require-SchemaVersion if
+						needed.
+					</li>
+
+				</ul>
+			</td>
+		</tr>
+		<tr>
+			<td class="table-header left-header">
+			MyCustomServiceUpgrade.java
+			</td>
+			<td class="">
+				<ul>
+					<li>
+						    Declare a UpgradeStepRegistrator component.
+					</li>
+					<li>
+					        It has no loggers.
+					</li>
+					<li>
+					        It has no reference to Release
+					        service.
```

Comment:
@mdelapenya, @mwilliams2014 - In the steps before this table, you need to explain what needs to be done to the Release service and why it needs to be done.